pkg/image/filefmt_test.go

```diff
@@ -46,118 +40,72 @@ func TestHeader_Match(t *testing.T) {
 	type args struct {
 		b []byte
 	}
-	tests := []struct {
-		name   string
-		fields fields
-		args   args
-		want   bool
-	}{
-		{
-			name:   "match gz",
-			fields: fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
-			args:   args{[]byte{0x1F, 0x8B}},
-			want:   true,
-		},
-		{
-			name:   "match qcow2",
-			fields: fields{"qcow2", []byte{'Q', 'F', 'I', 0xfb}, 0, 24, 8},
-			args:   args{[]byte{'Q', 'F', 'I', 0xfb}},
-			want:   true,
-		},
-		{
-			name:   "match tar",
-			fields: fields{"tar", []byte{0x75, 0x73, 0x74, 0x61, 0x72, 0x20}, 0x101, 124, 8},
-			args:   args{tarbyte},
-			want:   true,
-		},
-		{
-			name:   "match xz",
-			fields: fields{"xz", []byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}, 0, 0, 0},
-			args:   args{[]byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}},
-			want:   true,
-		},
-		{
-			name:   "failed match",
-			fields: fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
-			args:   args{[]byte{'Q', 'F', 'I', 0xfb}},
-			want:   false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			h := Header{
-				Format:      tt.fields.Format,
-				magicNumber: tt.fields.magicNumber,
-				mgOffset:    tt.fields.mgOffset,
-				SizeOff:     tt.fields.SizeOff,
-				SizeLen:     tt.fields.SizeLen,
-			}
-			if got := h.Match(tt.args.b); got != tt.want {
-				t.Errorf("Header.Match() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
 
-func TestHeader_Size(t *testing.T) {
-	type fields struct {
-		Format      string
-		magicNumber []byte
-		mgOffset    int
-		SizeOff     int
-		SizeLen     int
-	}
+	table.DescribeTable("Header match", func(fields fields, args args, want bool) {
+		h := Header{
+			Format:      fields.Format,
+			magicNumber: fields.magicNumber,
+			mgOffset:    fields.mgOffset,
+			SizeOff:     fields.SizeOff,
+			SizeLen:     fields.SizeLen,
+		}
+		got := h.Match(args.b)
+		Expect(got).To(Equal(want))
+	},
+		table.Entry("match gz",
+			fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
+			args{[]byte{0x1F, 0x8B}},
```

Comment:
args is a struct with a single byte array as a field, why not just pass the byte array?